scraper/scraper_bot.py

```diff
@@ -334,42 +451,104 @@ def _get_messages(self, after_message_id: str) -> List[HFDatasetScheme]:
             if obj not in unique_objects:
                 unique_objects.add(obj)
                 unique_list.append(obj)
+
         unique_list.sort(key=lambda x: x.message_id)
+        print(
+            f"Found {len(unique_list)} valid samples out of {total_messages} messages."
+        )
+        print(f"Oldest: {unique_list[0].timestamp}")
+        print(f"Newest: {unique_list[-1].timestamp}")
+
         return unique_list
 
-    def scrape(self, fetch_all: bool = False) -> Dataset:
-        (chunk, chunk_num) = self._load_dataset()
-        if chunk is None:
-            print("No existing dataset found.")
-            chunk = pd.DataFrame(columns=self.schema)
-            chunk_num = 0
-            after_message_id = None
-        else:
-            after_message_id = get_latest_message_id(chunk) if not fetch_all else None
-        messages = self._get_messages(after_message_id=after_message_id)
-        messages = self.filter_messages(messages) if fetch_all else messages
-        messages = prepare_dataset(messages)
+    def scrape(self, fetch_all: bool = False) -> None:
+        schema = [f.name for f in fields(HFDatasetScheme)]
+
+        # Drop images if we're not embedding them
+        if not self.embed_images:
+            schema.remove("image")
 
-        if not len(messages):
+        print(
+            f"Beginning scrape for {self.hf_dataset_name} with schema {schema} and fetch_all={fetch_all}"
+        )
+
+        # Load the current dataset without images initially, to figure out what we're working with
+        current_dataset, chunk_count = self._load_dataset(schema=schema)
```

Comment:
We **are** loading with images, since `schema = [f.name for f in fields(HFDatasetScheme)]` is at the start of the scrape function. Or am I missing something?

Reply:
Yea I forgot to update this part for github actions (still need to test that), but this saved me time trying to figure it out 😂